src/Microsoft.ManagedZLib/tests/DeflateStreamTests.cs

```diff
@@ -0,0 +1,96 @@
+        yield return new object[] { Path.Combine("UncompressedTestFiles", "plrabn12.txt") };
+        yield return new object[] { Path.Combine("UncompressedTestFiles", "ptt5") };
+        yield return new object[] { Path.Combine("UncompressedTestFiles", "sum") };
+        yield return new object[] { Path.Combine("UncompressedTestFiles", "xargs.1") };
+    }
+
+    public static IEnumerable<object[]> ByteArrayData()
+    {
+        yield return new object[] { new byte[] { 0x4C, 0x4C, 0x4F, 0x52, 0x41 } };
+        yield return new object[] {  new byte[] { 0x4C, 0x4C, 0x4F, 0x52, 0x41, 0x4C, 0x4C, 0x4F, 0x52, 0x41, 0x4C, 0x4C, 0x4F, 0x52, 0x41,
+                                            0x4C, 0x4C, 0x4F, 0x52, 0x41, 0x4C, 0x4C, 0x4F, 0x52, 0x41, 0x4C, 0x4C, 0x4F, 0x52, 0x41,
+                                            0x4C, 0x4C, 0x4F, 0x52, 0x41, 0x4C, 0x4C, 0x4F, 0x52, 0x41, 0x4C, 0x4C, 0x4F, 0x52, 0x41,
+                                            0x4C, 0x4C, 0x4F, 0x52, 0x41, 0x4C, 0x4C, 0x4F, 0x52, 0x41, 0x4C, 0x4C, 0x4F, 0x52, 0x41,
+                                            0x4C, 0x4C, 0x4F, 0x52, 0x41, 0x4C, 0x4C, 0x4F, 0x52, 0x41, 0x4C, 0x4C, 0x4F, 0x52, 0x41,
+                                            0x4C, 0x4C, 0x4F, 0x52, 0x41, 0x4C, 0x4C, 0x4F, 0x52, 0x41, 0x4C, 0x4C, 0x4F, 0x52, 0x41,
+                                            0x4C, 0x4C, 0x4F, 0x52, 0x41, 0x4C, 0x4C, 0x4F, 0x52, 0x41, 0x4C, 0x4C, 0x4F, 0x52, 0x41,
+                                            0x4C, 0x4C, 0x4F, 0x52, 0x41, 0x4C, 0x4C, 0x4F, 0x52, 0x41, 0x4C, 0x4C, 0x4F, 0x52, 0x41} };
+    }
+
+    [Theory]
+    [MemberData(nameof(ByteArrayData))]
+    public void ReadSmallArrays(byte[] bytes)
+    {
+        using MemoryStream originalData = new MemoryStream();
+        originalData.Write(bytes);
+        originalData.Position = 0;
+        VerifyRead(originalData);
+    }
+
+    [Theory]
+    [MemberData(nameof(UncompressedTestFiles))]
+    public void ReadFile(string path)
+    {
+        using FileStream fileStream = File.OpenRead(path);
+        VerifyRead(fileStream);
+    }
+
+    private void VerifyRead (Stream originalStream)
+    {
+        MemoryStream compressedDestination = new(); // Compressed data (with System.IO.Compresion)
+        using (System.IO.Compression.DeflateStream compressor = new System.IO.Compression.DeflateStream(compressedDestination, System.IO.Compression.CompressionMode.Compress, leaveOpen: true))
+        {
+            originalStream.CopyTo(compressor);  //Compressor has the compressed data now.
+        }
+        compressedDestination.Position = 0;
+
+        // Decompression
+        MemoryStream expectedStream = new();
+        using (DeflateStream decompressor = new DeflateStream(compressedDestination, CompressionMode.Decompress, leaveOpen: true))
+        {
+            decompressor.CopyTo(expectedStream);
+        }
+
+        using StreamReader readerOriginal = new StreamReader(originalStream);
```

Comment:
This helper method is used by both tests: the one that takes an array of bytes, and the one that takes a file.

The file ones were failing to pass because when comparing original vs expected as raw bytes. I suspect it has something to do with the way encoding works and how it gets compressed and then decompressed. But when we were comparing them as text strings, the tests passed.

What we could do is add an if statement here to decide if we want to compare the before and after as raw bytes or as text.